token-client/src/main/java/com/sap/cloud/security/client/DefaultHttpClientFactory.java

```diff
@@ -63,41 +58,42 @@ public CloseableHttpClient createClient(ClientIdentity clientIdentity) throws Ht
 			LOGGER.warn("Application has already created HttpClient for clientId = {}, please check.", clientId);
 		}
 		httpClientsCreated.add(clientId);
-
-		if (clientId == null || !clientIdentity.isCertificateBased()) {
-			LOGGER.warn(
-					"In productive environment provide well configured HttpClientFactory service, don't use default http client");
-			return HttpClients.custom().disableRedirectHandling().build();
-		} else {
-			return createTLSClient(clientIdentity);
+		if (clientId != null && clientIdentity.isCertificateBased()) {
+			SslConnection connectionPool = sslConnectionPool.computeIfAbsent(clientId,
+					s -> new SslConnection(clientIdentity));
+			return HttpClients.custom()
+					.setDefaultRequestConfig(requestConfig)
+					.setConnectionManager(connectionPool.poolingConnectionManager)
+					.setSSLContext(connectionPool.context)
+					.setSSLSocketFactory(connectionPool.sslSocketFactory)
+					.build();
 		}
+		LOGGER.warn(
```

Comment:
Not sure if we want to remove the warning in this place as well. If we want to keep it, please note that it is currently not printed for the TLS case above. Not sure if that is intended.